api/Events/User/ClientWantsToUpdateProfile.cs

```diff
@@ -16,23 +16,47 @@ public class ClientWantsToUpdateUserDto : BaseDtoWithJwt
 }
 
 [ValidateDataAnnotations]
-public class ClientWantsToUpdateProfile (UserService userService) : BaseEventHandler<ClientWantsToUpdateUserDto>
+public class ClientWantsToUpdateProfile (UserService userService, JwtService jwtService) : BaseEventHandler<ClientWantsToUpdateUserDto>
 {
     public override async Task Handle(ClientWantsToUpdateUserDto dto, IWebSocketConnection socket)
     {
-        var getUserDto = await userService.UpdateUser(dto.UpdateUserDto);
-        if (getUserDto == null)
+        var email = jwtService.GetEmailFromJwt(dto.Jwt);
+        var getUserDto = await userService.UpdateUser(dto.UpdateUserDto, email);
+        
+        if (getUserDto != null)
         {
-            throw new AppException("Failed to update user.");
+            socket.SendDto(new ServerConfirmsUpdate
+            {
+                GetUserDto = getUserDto
+            });
         }
-        socket.SendDto(new ServerConfirmsUpdate
+        else
         {
-            GetUserDto = getUserDto
-        });
+            var user = await userService.GetUserByEmail(email);
+            
+            if (user == null) throw new NotFoundException("User not found");
+            
+            socket.SendDto(new ServerRejectsUpdate
```

Comment:
if you throw an exception in here, socket.sendDto won't go through, as the exception will be caught by GlobalExceptionFilter and send out ServerRespondsNotFound

Reply:
that was the desired outcome, because if there is no user there isnt any data to return, but i am open to a different approach

Reply:
all right!